types/rest/rest.go

```diff
@@ -222,9 +223,17 @@ func ParseQueryHeightOrReturnBadRequest(w http.ResponseWriter, cliCtx context.CL
 }
 
 // PostProcessResponse performs post processing for a REST response.
-func PostProcessResponse(w http.ResponseWriter, cliCtx context.CLIContext, response interface{}) {
+// If the height is greater than zero it will be injected into the body
+// of the response. An internal server error is written to the response
+// if the height is negative or an encoding/decoding error occurs.
+func PostProcessResponse(w http.ResponseWriter, cliCtx context.CLIContext, response interface{}, height int64) {
```

Comment:
I think it'd be ideal to keep this function signature the same. We already have `cliCtx` being passed in. Prior to calling `PostProcessResponse`, simply set the new ctx (i.e. `ctx = ctx.WithHeight(height)`).